nicos_ess/loki/gui/sampleconf.py

```diff
@@ -278,11 +279,18 @@ def on_client_disconnected(self):
 
     def setViewOnly(self, viewonly):
         for control in [
-            self.createBtn, self.retrieveBtn, self.openFileBtn, self.applyBtn,
+            self.createBtn, self.retrieveBtn, self.openFileBtn,
             self.saveBtn, self.newBtn, self.editBtn, self.delBtn, self.frame,
             self.list
         ]:
             control.setEnabled(not viewonly)
+        # Handle apply button separately.
+        if viewonly:
+            self.applyBtn.setEnabled(False)
+        # If one toggles view only mode without applying changes, upon exiting
+        # view-only mode, following ensures apply button is enabled.
+        elif self.dirty and not viewonly:
```

Comment:
```suggestion
        elif self.dirty:
```
if it is viewonly then it won't get to this clause

Reply:
Yes that is my intention. If it is viewonly apply button goes away. But after you toggle back from view only, it remembers it is dirty and enables apply.

Reply:
It is what I have written to the comment, basically.

Reply:
If viewonly=True then it will never reach line 292, if it does reach line 292 then viewonly must be False, so the check is superfluous.